Test/Sample data fails to load during Vagrant deploy

Newman fails to compile due to a gyp error, looks like contextify uses a NaN API that's "not there yet" in node v0.10? This prevents the sample data from being loaded.

Not sure what would've caused this.

Relevant log entries:
```
==> ozp-centos6.6: make: Entering directory `/usr/lib/node_modules/newman/node_modules/jsdom/node_modules/contextify/build'
==> ozp-centos6.6:   CXX(target) Release/obj.target/contextify/src/contextify.o
==> ozp-centos6.6: In file included from ../src/contextify.cc:3:
==> ozp-centos6.6: ../node_modules/nan/nan.h:328: error: ‘REPLACE_INVALID_UTF8’ is not a member of ‘v8::String’
==> ozp-centos6.6: make: *** [Release/obj.target/contextify/src/contextify.o] Error 1
==> ozp-centos6.6: make: Leaving directory `/usr/lib/node_modules/newman/node_modules/jsdom/node_modules/contextify/build'
==> ozp-centos6.6: gyp ERR! build error 
==> ozp-centos6.6: gyp ERR! stack Error: `make` failed with exit code: 2
==> ozp-centos6.6: gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/node-gyp/lib/build.js:267:23)
==> ozp-centos6.6: gyp ERR! stack     at ChildProcess.emit (events.js:98:17)
==> ozp-centos6.6: gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:820:12)
==> ozp-centos6.6: gyp ERR! System Linux 2.6.32-573.el6.x86_64
==> ozp-centos6.6: gyp ERR! command "node" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
==> ozp-centos6.6: gyp ERR! cwd /usr/lib/node_modules/newman/node_modules/jsdom/node_modules/contextify
==> ozp-centos6.6: gyp ERR! node -v v0.10.36
==> ozp-centos6.6: gyp ERR! node-gyp -v v0.10.6
==> ozp-centos6.6: gyp ERR! not ok 
==> ozp-centos6.6: npm
==> ozp-centos6.6:  
==> ozp-centos6.6: ERR!
==> ozp-centos6.6:  
==> ozp-centos6.6: weird error
==> ozp-centos6.6:  1
==> ozp-centos6.6: npm
==> ozp-centos6.6:  
==> ozp-centos6.6: ERR!
==> ozp-centos6.6:  
==> ozp-centos6.6: not ok
==> ozp-centos6.6:  code 0
...
==> ozp-centos6.6: /vagrant/deploy.sh: line 60: newman: command not found
==> ozp-centos6.6: /vagrant/deploy.sh: line 61: newman: command not found
```

--- a/vagrant/centos6.6/build-and-deploy-box/build.sh
+++ b/vagrant/centos6.6/build-and-deploy-box/build.sh
@@ -39,6 +39,9 @@ printf "\n**************\n  finished cloning front-end repos \n*************\n"
 # http://stackoverflow.com/questions/22387857/stop-bower-from-asking-for-statistics-when-installing
 export CI=true
 
+source ${HOMEDIR}/.nvm/nvm.sh
+nvm use default
+
 # - - - - - - - - - - - - - - - - - - -
 # 				Webtop
 # - - - - - - - - - - - - - - - - - - -

--- a/vagrant/centos6.6/build-and-deploy-box/deploy.sh
+++ b/vagrant/centos6.6/build-and-deploy-box/deploy.sh
@@ -57,6 +57,9 @@ sed -i "s/http:\/\/ozone-development.github.io\/ozp-demo/https:\/\/${HOST_IP}:77
 sed -i "s/http:\/\/ozone-development.github.io\/iwc\/owf7adapter.html?url=http%3A%2F%2Fozone-development.github.io%2Fozp-demo/https:\/\/${HOST_IP}:7799\/iwc\/owf7adapter.html?url=https%3A%2F%2F${HOST_IP}%3A7799%2Fdemo_apps/g" postman/data/modifiedListingData.json
 printf "Sleeping for 2 minutes waiting for server to start"
 sleep 2m
+
+source ${HOMEDIR}/.nvm/nvm.sh
+nvm use default
 newman -k -c postman/createSampleMetaData.json -e postman/env/localDev.json
 newman -k -c postman/createSampleListings.json -e postman/env/localDev.json -n 34 -d postman/data/modifiedListingData.json
 printf "\n*****************\n  Finished deploying backend \n*****************\n"

--- a/vagrant/centos6.6/build-and-deploy-box/dev-scripts/test-center-branch.sh
+++ b/vagrant/centos6.6/build-and-deploy-box/dev-scripts/test-center-branch.sh
@@ -35,6 +35,8 @@ git checkout ${BRANCH_NAME} | tee -a ${OUTPUT_FILE}
 export CI=true
 rm -rf node_modules
 rm -rf bower_components
+source ${HOMEDIR}/.nvm/nvm.sh
+nvm use default
 npm install | tee -a ${OUTPUT_FILE}
 bower install | tee -a ${OUTPUT_FILE}
 npm run build | tee -a ${OUTPUT_FILE}

--- a/vagrant/centos6.6/build-and-deploy-box/dev-scripts/webtop-rebuild-redeploy.sh
+++ b/vagrant/centos6.6/build-and-deploy-box/dev-scripts/webtop-rebuild-redeploy.sh
@@ -30,6 +30,8 @@ export CI=true
 cd ${RSYNC_DIR}/ozp-webtop
 rm -rf node_modules
 rm -rf vendor
+source ${HOMEDIR}/.nvm/nvm.sh
+nvm use default
 npm install
 bower install
 npm run build

--- a/vagrant/centos6.6/build-and-deploy-box/initial_provisioning.sh
+++ b/vagrant/centos6.6/build-and-deploy-box/initial_provisioning.sh
@@ -15,7 +15,11 @@ sudo yum install kernel-headers kernel-devel -y
 printf "\n*************\n finished installing kernel headers \n*************\n"
 sudo yum install java-1.7.0-openjdk java-1.7.0-openjdk-devel git unzip vim -y
 printf "\n**************\n finished installing java and such \n**************\n"
-sudo yum install nodejs npm -y
+sudo yum install gcc-c++ -y
+curl -s https://raw.githubusercontent.com/creationix/nvm/v0.30.1/install.sh | bash
+source ${HOMEDIR}/.nvm/nvm.sh
+nvm install 0.12
+nvm alias default node
 printf "\n**************\nfinished installing nodejs and npm\n**************\n"
 # install gvm
 curl -s get.gvmtool.net | bash

--- a/vagrant/centos6.6/build-and-deploy-box/initial_provisioning.sh
+++ b/vagrant/centos6.6/build-and-deploy-box/initial_provisioning.sh
@@ -76,7 +76,7 @@ sudo yum remove mysql mysql-* -y
 sudo yum --enablerepo=remi,remi-test install mysql mysql-server tomcat elasticsearch nginx multitail -y
 
 # Install newman (for adding test data)
-sudo npm install -g newman bower
+npm install -g newman bower
 
 printf "\n***************\nfinished deployment installation\n***************\n"
 

